test/json_unittest.cc

```diff
@@ -742,12 +937,48 @@ TEST(StreamingTest, MultipleObjectsStream) {
 }
 
 TEST(StreamingTest, ParseStreamReturnsByteCount) {
+  GuardJsonException([](){
+    json::value v;
+    json::Parser p([&v](json::value r) { v = std::move(r); });
+    size_t n = p.ParseStream(std::istringstream("[1]"));
+    EXPECT_TOSTRING_EQ("[1.0]", v);
+    EXPECT_EQ(3, n);
+  });
+}
+
+TEST(StreamingTest, ParseStreamNeedsEofToParseNumbers) {
   GuardJsonException([](){
     json::value v;
     json::Parser p([&v](json::value r) { v = std::move(r); });
     size_t n = p.ParseStream(std::istringstream("123"));
-    EXPECT_TOSTRING_EQ("123", v);
     EXPECT_EQ(3, n);
+    EXPECT_EQ(nullptr, v);
+    // The number is not known until EOF is reached.
+    p.NotifyEOF();
+    EXPECT_TOSTRING_EQ("123.0", v);
+  });
+}
+
+TEST(StreamingTest, ParseStreamThrowsOnParseErrorMidStream) {
```

Comment:
I find this name (ParseStreamThrowsOnParseErrorMidStream) hard to understand. 
Removing the "On" would be better.

Reply:
Unfortunately, that would change the meaning I was trying to convey. Renamed — is this better?